src/applications/disability-benefits/all-claims/tests/components/ToxicExposureChoicePage.unit.spec.jsx

```diff
@@ -0,0 +1,867 @@
+
+      await waitFor(() => {
+        expect(goForward.calledOnce).to.be.true;
+        expect(goForward.calledWith(data)).to.be.true;
+      });
+    });
+
+    it('should allow navigation when no condition is selected and no existing toxic exposure data', async () => {
+      const data = {
+        newDisabilities: [{ condition: 'Chronic Bronchitis' }],
+        toxicExposure: {
+          conditions: {},
+        },
+      };
+
+      const { goForward, user } = renderPage({ data });
+      const submitButton = document.querySelector('button[type="submit"]');
+
+      await user.click(submitButton);
+
+      await waitFor(() => {
+        expect(goForward.calledOnce).to.be.true;
+      });
+    });
+
+    it('should show modal when no condition is selected but has existing toxic exposure data', async () => {
+      const data = {
+        newDisabilities: [{ condition: 'Chronic Bronchitis' }],
+        toxicExposure: {
+          conditions: {},
+          gulfWar1990: {
+            bahrain: true,
+          },
+        },
+      };
+
+      const { goForward, user } = renderPage({
+        data,
+        featureToggleEnabled: true,
+      });
+      const submitButton = document.querySelector('button[type="submit"]');
+
+      await user.click(submitButton);
+
+      await waitFor(() => {
+        const modal = document.querySelector('va-modal');
+        expect(modal).to.exist;
+        expect(modal.getAttribute('visible')).to.equal('true');
+        expect(goForward.called).to.be.false;
+      });
+    });
+
+    it('should allow navigation when selecting "none" option and no existing data', async () => {
+      const data = {
+        newDisabilities: [{ condition: 'Chronic Bronchitis' }],
+        toxicExposure: {
+          conditions: {
+            none: true,
+          },
+        },
+      };
+
+      const { goForward, user } = renderPage({ data });
+      const submitButton = document.querySelector('button[type="submit"]');
+
+      await user.click(submitButton);
+
+      await waitFor(() => {
+        expect(goForward.calledOnce).to.be.true;
+      });
+    });
+
+    it('should show modal when selecting "none" option but has existing toxic exposure data', async () => {
+      const data = {
+        newDisabilities: [{ condition: 'Chronic Bronchitis' }],
+        toxicExposure: {
+          conditions: {
+            none: true,
+          },
+          herbicide: {
+            vietnam: true,
+          },
+        },
+      };
+
+      const { goForward, user } = renderPage({
+        data,
+        featureToggleEnabled: true,
+      });
+      const submitButton = document.querySelector('button[type="submit"]');
+
+      await user.click(submitButton);
+
+      await waitFor(() => {
+        const modal = document.querySelector('va-modal');
+        expect(modal).to.exist;
+        expect(modal.getAttribute('visible')).to.equal('true');
+        expect(goForward.called).to.be.false;
+      });
+    });
+  });
+
```

Comment:
What happens in the case where a condition and "none" are selected, whether or not there are existing toxic exposure data?